tests/migration/version_switch_upgrade_target.pm

```diff
@@ -50,6 +50,9 @@ sub run {
     # KDE has moved to Package Hub, it will stay install with SLES15 if added PackageHub
     set_var('DM_NEEDS_USERNAME', '1') if (check_var('DESKTOP', 'KDE') && is_sle('15+') && (get_var('ADDONURL', '') !~ /phub/));
 
+    # Reset DESKTOP after upgrade as desktop change
+    set_var('DESKTOP', 'gnome') if (is_sle('<12', get_var('HDDVERSION')) && check_var('DESKTOP', 'kde'));
```

Comment:
There is already a very similar condition a line up, so that can be reused:

```
# In SLE 15, KDE was dropped, after migration the default desktop is GNOME
# KDE has moved to Package Hub, it will stay installed with SLE 15 if PackageHub is available
if (check_var('DESKTOP', 'KDE') && is_sle('15+') && (get_var('ADDONURL', '') !~ /phub/)) {
    set_var('DM_NEEDS_USERNAME', '1');
    set_var('DESKTOP', 'gnome');
}
```

I'm a bit confused by the two assignments to `DM_NEEDS_USERNAME` which can conflict with each other and also the comment which says `after migration the default desktop is XDM`, which doesn't seem to be true.

Reply:
This is a little different with my condition. For my case, After upgrade, it may upgrade to sles12 or sles15... I only need check base system is 11sp4 with KDE. After upgrade, the desktop will change to gnome.  For "if (check_var('DESKTOP', 'KDE') && is_sle('15+') && (get_var('ADDONURL', '') !~ /phub/))" this condition. It means the desktop will stay to use 'KDE', it won't change to gnome

Reply:
my understand: For desktop is KDE, it need set DM_NEEDS_USERNAME=1. But after upgrade, the desktop changed to gnome, it should set DM_NEEDS_USERNAME=0.

Reply:
> For my case, After upgrade, it may upgrade to sles12 or sles15...

Are there still migration tests from 11SP4+KDE to SLE12? AFAICT that would also switch to gnome, so setting `DESKTOP` to `gnome` in that case would still be correct. Just changing `is_sle('15+')` to `is_sle('12+')` would be needed then.

Reply:
I checked the case and code. Those code not used any more. So, I deleted those code and change the desktop setting to the first condition.

Reply:
> I checked the case and code. Those code not used any more. So, I deleted those code and change the desktop setting to the first condition.

Are you sure? If you upgrade from 11 SP4 to 15 with packagehub enabled, you won't end up with GNOME.

Reply:
I add the condition to check whether sles15 include packagehub module.

Reply:
I add the condition to check whether sles15 include packagehub module.